libs/core/algorithms/tests/performance/benchmark_scan_algorithms.cpp

```diff
@@ -0,0 +1,213 @@
+//  Copyright (c) 2021 Akhil J Nair
+//
+//  SPDX-License-Identifier: BSL-1.0
+//  Distributed under the Boost Software License, Version 1.0. (See accompanying
+//  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+#include <hpx/local/init.hpp>
+#include <hpx/modules/program_options.hpp>
+#include <hpx/modules/testing.hpp>
+#include <hpx/parallel/algorithms/copy.hpp>
+#include <hpx/parallel/algorithms/exclusive_scan.hpp>
+#include <hpx/parallel/algorithms/inclusive_scan.hpp>
+#include <hpx/parallel/algorithms/transform_exclusive_scan.hpp>
+#include <hpx/parallel/algorithms/transform_inclusive_scan.hpp>
+#include <hpx/parallel/algorithms/unique.hpp>
+
+#include <array>
+#include <cstddef>
+#include <fstream>
+#include <functional>
+#include <iostream>
+#include <map>
+#include <string>
+#include <vector>
+
+//#define OUTPUT_TO_CSV
+
+enum class ALGORITHM
+{
+    EXCLUSIVE_SCAN,
+    INCLUSIVE_SCAN,
+    TRANSFORM_EXCLUSIVE_SCAN,
+    TRANSFORM_INCLUSIVE_SCAN,
+    COPY_IF,
+    UNIQUE_COPY
+};
+
+void measureScanAlgorithms()
+{
+#if defined(OUTPUT_TO_CSV)
+    std::map<ALGORITHM, std::string> filenames = {
+        {ALGORITHM::EXCLUSIVE_SCAN, "exclusiveScanCompare.csv"},
+        {ALGORITHM::INCLUSIVE_SCAN, "inclusiveScanCompare.csv"},
+        {ALGORITHM::TRANSFORM_EXCLUSIVE_SCAN,
+            "transformExclusiveScanCompare.csv"},
+        {ALGORITHM::TRANSFORM_INCLUSIVE_SCAN,
+            "transformInclusiveScanCompare.csv"},
+        {ALGORITHM::COPY_IF, "copyIfCompare.csv"},
+        {ALGORITHM::UNIQUE_COPY, "uniqueCopyCompare.csv"},
+    };
+#endif
+
+    for (int alg = (int) ALGORITHM::EXCLUSIVE_SCAN;
+         alg <= (int) ALGORITHM::UNIQUE_COPY; alg++)
+    {
+        std::size_t start = 32;
+        std::size_t till = 1 << 28;
```

Comment:
@Jedi18 this benchmark is timing out in CI, and I'm guessing it's because of this being too large in debug mode. Could you check how long a debug run of this test runs, and adjust this (in debug mode only) to something smaller. Try to target having the test run no longer than 30 seconds.

Reply:
Thanks yeah it runs for too long, I've reduced the NUM_ITERATIONS and till variable, forgot to push the changes.

This PR is held up on a bug with copy_if that is causing the reduce_by_key test to fail, I'll look into it this week.